x-pack/plugins/enterprise_search/server/utils/create_pipeline_definitions.ts

```diff
@@ -225,3 +230,58 @@ export const createIndexPipelineDefinitions = (
   });
   return { created: [indexName, `${indexName}@custom`, `${indexName}@ml-inference`] };
 };
+
+/**
+ * Format the body of an ML inference pipeline for a specified model.
+ * Does not create the pipeline, only returns JSON for the user to preview.
+ * @param modelId modelId selected by user.
+ * @param sourceField The document field that model will read.
+ * @param destinationField The document field that the model will write to.
+ * @param esClient the Elasticsearch Client to use when retrieving model details.
+ */
+export const formatMlPipelineBody = async (
+  modelId: string,
+  sourceField: string,
+  destinationField: string,
+  esClient: ElasticsearchClient
+): Promise<ESPipeline> => {
+  const models = await esClient.ml.getTrainedModels({ model_id: modelId });
+  const model = models.trained_model_configs[0];
```

Comment:
question: Is this definitely defined? If there's a possibility that there are no trained models, we will need to check for that and handle it. Same for the field_names below.

Reply:
How do we want to handle that? Raise an exception? Return an error body? Return an empty dict or undefined?

Reply:
@sphilipse I added handling for edge cases - take a look.